src/Core/mcts/UCT.cpp

```diff
@@ -1,6 +1,57 @@
 #include <UCT.h>
 
-UCT::UCT(EnvironmentBase &environment, boost::function<int(State)> default_policy)
-    : MCTSBase(environment, default_policy){};
+UCT::UCT(EnvironmentBase &environment) : MCTSInterface(environment), generator(std::mt19937(time(nullptr))){};
 
-void UCT::run(int n_searches) {}
+std::shared_ptr<SearchNode> UCT::m_tree_policy(std::shared_ptr<SearchNode> node) { return m_tpolicy.treePolicy(node); };
+
+Reward UCT::m_default_policy(State &state) { return m_defaultPolicy.defaultPolicy(state); };
+
+std::shared_ptr<SearchNode> UCT::m_best_child(std::shared_ptr<SearchNode> node, double c) {
+    int best_child = 0;
+    double best_score_so_far = -1;
+
+    for (int i = 0; i < node->child_nodes.size(); i++) {
+        auto child = node->child_nodes.at(i);
+        double score = (child->score / (child->visits + 0.000000001)) +
+                       c * std::sqrt((std::log(node->visits) / (child->visits + 0.000000001)));
```

Comment:
Consider using `DBL_MIN` instead of 0.0000000001

Reply:
`DBL_MIN` is in `values.h>`

Reply:
On my machine:
`std::numeric_limits<double>::lowest() = -1.79769e+308`
`DBL_MIN = 2.22507e-308`